policy/release/olm.rego

```diff
@@ -215,6 +215,49 @@ deny contains result if {
 	result := lib.result_helper_with_term(rego.metadata.chain(), [image.str(unmatched_image.ref)], image.str(unmatched_image.ref))
 }
 
+# METADATA
+# title: Images referenced by OLM bundle are from allowed registries
+# description: >-
+#   Images referenced by the OLM bundle are from allowed registries
+# custom:
+#   short_name: allowed_registries
+#   failure_msg: The %q CSV image reference is not from an allowed registry.
+#   solution: >-
+#     The image must be hosted on an allowed registry.
+#   collections:
+#   - redhat
+#   effective_on: 2024-08-15T00:00:00Z
+#
+deny contains result if {
+	# Restrict to release pipeline for now
+	_release_restrictions_apply
```

Comment:
I think it may make sense, but I did this to be consistent with other work in the same Epic. I suggest we keep it at release for now and un-restrict this and other OLM checks in a future task.

Reply:
ack. Keep in mind that finding out about a violation at release time it's quite terrible. We would like to avoid this as much as possible. We should have a strong reason to make it release-only, not the other way around.

Reply:
@lcarva you make a good point. After consulting a couple folks, I will remove the restriction.

Reply:
Removed the restriction in the latest push